packages/wrapper-react/src/index.tsx

```diff
@@ -30,72 +29,49 @@ export const MonacoEditorReactComp: React.FC<MonacoEditorProps> = (props) => {
     const containerRef = useRef<HTMLDivElement>(null);
 
     useEffect(() => {
-        return () => {
-            destroyMonaco();
+        const destroyMonaco = async () => {
+            try {
+                await wrapperRef.current.dispose();
+            } catch {
+                // The language client may throw an error during disposal.
+                // This should not prevent us from continue working.
+            }
         };
-    }, []);
-
-    useEffect(() => {
-        handleReInit();
-    }, [wrapperConfig]);
-
-    useEffect(() => {
-        handleOnTextChanged();
-    }, [onTextChanged]);
-
-    useEffect(() => {
-        if (containerRef.current) {
-            containerRef.current.className = className ?? '';
-            wrapperConfig.htmlContainer = containerRef.current;
-        }
-    }, [className]);
 
-    const handleReInit = useCallback(async () => {
-        await destroyMonaco();
-        await initMonaco();
-        await startMonaco();
-    }, [wrapperConfig]);
-
-    const initMonaco = useCallback(async () => {
-        if (containerRef.current) {
-            wrapperConfig.htmlContainer = containerRef.current;
-            await wrapperRef.current.init(wrapperConfig);
-        } else {
-            throw new Error('No htmlContainer found! Aborting...');
-        }
-    }, [wrapperConfig]);
+        const initMonaco = async () => {
```

Comment:
You can move all methods outside the use effects, so you can use them in either one.

Reply:
See above ⬆️